packages/components/src/components/stateful-pages/ResourceForm/ResourceForm.tsx

```diff
@@ -1,25 +1,68 @@
-import React from 'react'
+import React, { useState } from 'react'
 
 import { useResource, useResourceOperations } from '../../../hooks'
 
 import CustomForm from '../../organisms/CustomForm/CustomForm'
+import Button from '../../atoms/buttons/Button'
+import { isError } from '@elastic-suite/gally-admin-shared'
+import { closeSnackbar, enqueueSnackbar } from 'notistack'
+import { useTranslation } from 'react-i18next'
+import { styled } from '@mui/system'
+import { Box } from '@mui/material'
 
+const CustomResourceForm = styled('div')(({ theme }) => ({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: theme.spacing(2),
+}))
 interface IProps {
   // id?: string
   resourceName: string
 }
 
 function ResourceForm(props: IProps): JSX.Element {
+  const { t } = useTranslation('categories')
   const { resourceName } = props
   const resource = useResource(resourceName)
-  const { create } = useResourceOperations(resource)
+
+  const { create } = useResourceOperations<any>(resource)
+
+  const [isLoading, setIsLoading] = useState(false)
+  const requiredChamps = resource.supportedProperty.filter(
+    (property) => property.required && property?.gally?.visible
+  )
+
+  const [data, setData] = useState<Record<string, unknown>>() // boostData if it's for updateForm
```

Comment:
si ton type est un `Record` donne une valeur par défaut qui soit cohérente avec ce type:
```suggestion
  const [data, setData] = useState<Record<string, unknown>>({})
```
ça permet probablement d'éviter d'utiliser le `as` ainsi que le `?.` à la ligne 38